libbeat/outputs/elasticsearch/config.go

```diff
@@ -26,20 +26,21 @@ import (
 )
 
 type elasticsearchConfig struct {
-	Protocol         string            `config:"protocol"`
-	Path             string            `config:"path"`
-	Params           map[string]string `config:"parameters"`
-	Headers          map[string]string `config:"headers"`
-	Username         string            `config:"username"`
-	Password         string            `config:"password"`
-	APIKey           string            `config:"api_key"`
-	LoadBalance      bool              `config:"loadbalance"`
-	CompressionLevel int               `config:"compression_level" validate:"min=0, max=9"`
-	EscapeHTML       bool              `config:"escape_html"`
-	Kerberos         *kerberos.Config  `config:"kerberos"`
-	BulkMaxSize      int               `config:"bulk_max_size"`
-	MaxRetries       int               `config:"max_retries"`
-	Backoff          Backoff           `config:"backoff"`
+	Protocol           string             `config:"protocol"`
+	Path               string             `config:"path"`
+	Params             map[string]string  `config:"parameters"`
+	Headers            map[string]string  `config:"headers"`
+	Username           string             `config:"username"`
+	Password           string             `config:"password"`
+	APIKey             string             `config:"api_key"`
+	LoadBalance        bool               `config:"loadbalance"`
+	CompressionLevel   int                `config:"compression_level" validate:"min=0, max=9"`
+	EscapeHTML         bool               `config:"escape_html"`
+	Kerberos           *kerberos.Config   `config:"kerberos"`
+	BulkMaxSize        int                `config:"bulk_max_size"`
+	MaxRetries         int                `config:"max_retries"`
+	Backoff            Backoff            `config:"backoff"`
+	NonIndexablePolicy NonIndexablePolicy `config:"non_indexable_policy"`
```

Comment:
Given that we have two methods for handling events that cannot be indexed and not all options make sense the two actions should be separated a bit better to guide users through the collect configuration path.

Instead of letting users configure an action and if that action id `dead_letter_index` then read the index name, I would rather configure the option like this:

```yaml
output.elasticsearch:
  non_indexable_policy.drop: ~
```

```yml
output.elasticsearch:
  non_indexable_policy.dead_letter_index:
    index: "my-dead-letter-index"
```

Basically, you have to make `NonIndexablePolicy` a `common.ConfigNamespace` and create a configuration object based on the namespace. We use the same pattern in `file_identifier` option in Filebeat: https://github.com/elastic/beats/blob/master/filebeat/input/filestream/identifier.go#L79

Reply:
yes, this looks more consistent. I'll have a go at it.